numpy/f2py/tests/test_crackfortran.py

```diff
@@ -38,3 +39,19 @@ def test_module(self):
         assert_array_equal(k, w + 1)
         assert self.module.t0(23) == b'2'
 
+
+class TestMarkinnerspaces():
+    # issue #14118: markinnerspaces does not handle multiple quotations
+
+    def test_do_not_touch_normal_spaces(self):
+        test_list = ["a ", " a", "a b c", "'abcdefghij'"]
+        for i in test_list:
+            assert_equal(markinnerspaces(i), i)
+
+    def test_one_relevant_space(self):
+        assert_equal(markinnerspaces("a 'b c' \\\' \\\'"), "a 'b@_@c' \\' \\'")
+        assert_equal(markinnerspaces("a \"b c\" \\\" \\\""), "a \"b@_@c\" \\\" \\\"")
```

Comment:
```suggestion
        assert_equal(markinnerspaces(r'a "b c" \" \"'), r'a "b@_@c" \" \"')
```
Just a thought to make it more readable, the `r` string part applies also above.

Are confusion about multiple escape characters following each other OK in this function, or is that (another) lingering bug?

Reply:
Probably also a lingering bug here if the string already contains the `@_@` marker

Reply:
I agree, @eric-wieser, the bug is there. As a fix, the `@_@` marker could be replaced with a different marker, say `@F2PY_INNERSPACE_F2PY@`, that would make the bug to show up less likely.

Reply:
Since `@{}@` markers are used through this file, probably better to use `@F2PY_PARSING_MARKER{}@` instead, where `{}` would be `_` here.

Let's not worry about that in this patch though